### ISSUE TITLE
stuck button

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#148]
Mav and Ron got heads-up and Mav got the button every hand.  WTF? That definitely used to work. I'll have to look at when I last changed any code near there. I know I was looking at the code in anticipation of people coming in from other tables between the blinds, but I don't think I touched anything.

```
maverick600 blinds 4000
JADC blinds 8000
2 B   deadhead      10300    0  [  ] [  ]
3     maverick600   36550 4000  [  ] [  ]
7     JADC          11600 8000  [  ] [  ]
8   > Schmuckbox   124050    0  [  ] [  ]
...
maverick600    Ts 7s     Pair of Deuces 2c 2s Ah Ts 9c
JADC           5d 6d     Pair of Deuces 2c 2s Ah 9c 6d
maverick600 wins 39200

JADC bust in 4th place
Dealing 8000 16000 Texas Hold’em

Schmuckbox blinds 8000
2   > deadhead      10300    0  [  ] [  ]
3 B   maverick600   60150    0  [  ] [  ]
8     Schmuckbox   116050 8000  [  ] [  ]
...
Dealing 8000 16000 Texas Hold’em

Schmuckbox blinds 4000
deadhead blinds 8000
2     deadhead      10300 8000  [  ] [  ]
3   > maverick600   60150    0  [  ] [  ]
8     Schmuckbox   112050 4000  [  ] [  ]
deadhead       8d Ad     Ace High Flush           Ad Kd Jd Td 8d
maverick600    Qd 9d     King High Straight Flush Kd Qd Jd Td 9d
maverick600 wins 40600

deadhead bust in 3rd place, winning: 200
Dealing 8000 16000 Texas Hold’em

maverick600 blinds 8000
3     maverick600   74450 8000  [  ] [  ]
8 B > Schmuckbox   112050    0  [  ] [  ]
...
Dealing 8000 16000 Omaha High/Low Eight-or-Better

maverick600 blinds 4000
Schmuckbox blinds 8000
3   > maverick600   78450 4000  [  ] [  ] [  ] [  ]
8     Schmuckbox   104050 8000  [  ] [  ] [  ] [  ]
...
Dealing 8000 16000 Omaha High/Low Eight-or-Better

maverick600 blinds 4000
Schmuckbox blinds 8000
3 B > maverick600   74450 4000  [  ] [  ] [  ] [  ]
8     Schmuckbox   108050 8000  [  ] [  ] [  ] [  ]
...
Dealing 8000 16000 Omaha High/Low Eight-or-Better

maverick600 blinds 4000
Schmuckbox blinds 8000
3 B > maverick600   70450 4000  [  ] [  ] [  ] [  ]
8     Schmuckbox   112050 8000  [  ] [  ] [  ] [  ]
...
Dealing 8000 16000 Omaha High/Low Eight-or-Better

maverick600 blinds 4000
Schmuckbox blinds 8000
...
Dealing 10000 20000 Omaha High/Low Eight-or-Better

maverick600 blinds 5000
Schmuckbox blinds 10000
3 B > maverick600  101450  5000  [  ] [  ] [  ] [  ]
8     Schmuckbox    78050 10000  [  ] [  ] [  ] [  ]
...
Dealing 10000 20000 Omaha High/Low Eight-or-Better

maverick600 blinds 5000
Schmuckbox blinds 10000
3 B > maverick600  96450  5000  [  ] [  ] [  ] [  ]
8     Schmuckbox   83050 10000  [  ] [  ] [  ] [  ]
...
Dealing 10000 20000 Omaha High/Low Eight-or-Better

maverick600 blinds 5000
Schmuckbox blinds 10000
3 B > maverick600  96450  5000  [  ] [  ] [  ] [  ]
8     Schmuckbox   83050 10000  [  ] [  ] [  ] [  ]
...
Dealing 10000 20000 Omaha High/Low Eight-or-Better

maverick600 blinds 5000
Schmuckbox blinds 10000
3 B > maverick600  91450  5000  [  ] [  ] [  ] [  ]
8     Schmuckbox   88050 10000  [  ] [  ] [  ] [  ]
...
Dealing 10000 20000 (3000) Stud
Everyone antes 2000
3 B   maverick600   89450 2000  [  ] [  ] Ts
8   > Schmuckbox   101050 2000  [  ] [  ] 4c
Schmuckbox brings-in for 3000
```

**Comment by ctm** [comment]
Perhaps I should be pilloried for guessing in comments, but... this appears to be related to the third place player busting when in the big blind, leaving no small blind the next hand.  However, that should happen more than a third of the time and AFAIK, nobody has seen this behavior before, so I figure it must be a "new" but. Perhaps this commit is responsible:
```
commit a9fa79467cbdf67d9d667d5b6c197b05312d43d2
Author: Clifford T. Matthews <ctm@devctm.com>
Date:   Mon Mar 23 19:31:15 2020 -0600

    [Bug] won't try to get blinds from seats without players.  Fixes #110
```
AFAICT, it's the most recent change to that section of code. That's far enough back that I would think we'd have seen it by now, however, we're typically only playing a single tournament a day and some of them are straight up stud tournaments and others are mix games. So it's certainly within the realm of possibility that the bug was introduced on March 23rd but only noticed yesterday.

**Comment by ctm** [comment]
Nope. Turns out the table was caching the number of players at the table but only using that cached info to determine whether to do special blind handling when it gets down to two players. However, that cached value wasn't being updated when players were moved in. We only hit that code once per hand, so the cache really wasn't worth the effort, so I got rid of it.

I'm deploying this now. I still have to get at least one more deploy out today since I haven't yet added WTB PL HE.
